File: Scripts/laptop/palindrome-detection/go/palindrome-detection.go
package main

func executeTask(i int) int{

	var flag bool
	flag = isPalindrome("saippuakivikauppias")
	_ = flag
	return i + 1
}

func main() {
	r := int(1)
	for i := 0; i < 1000000000; i++ {
		r = executeTask(i + r)	
	}
}

func isPalindrome(s string) bool {
	runes := []rune(s)
	for len(runes) > 1 {
		if runes[0] != runes[len(runes)-1] {
			return false
		}
		runes = runes[1 : len(runes)-1]
	}
	return true
}

File: Scripts/laptop/palindrome-detection/c++/palindrome-detection.cpp
#include <string>
#include <algorithm>

bool is_palindrome(std::string);

int executeTask(int i) {
	
	std::string array = "saippuakivikauppias" + (char)i;
	is_palindrome(array);
	return i;
}

int main() {

	volatile int r;
	for (int i=0; i<2000000000; ++i) {
		r = executeTask(i );
	}
	return 0;
}


bool is_palindrome(std::string s)
{
  return std::equal(s.begin(), s.end(), s.rbegin());
}
